x-pack/plugins/snapshot_restore/public/app/sections/home/snapshot_list/snapshot_table/snapshot_table.tsx

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { RouteComponentProps, withRouter } from 'react-router-dom';
+
+import { EuiButton, EuiInMemoryTable, EuiLink } from '@elastic/eui';
+
+import { SnapshotDetails } from '../../../../../../common/types';
+import { useAppDependencies } from '../../../../index';
+import { formatDate } from '../../../../services/text';
+
+interface Props extends RouteComponentProps {
+  snapshots: SnapshotDetails[];
+  reload: () => Promise<void>;
+  openSnapshotDetails: (repositoryName: string, snapshotId: string) => void;
+}
+
+const SnapshotTableUi: React.FunctionComponent<Props> = ({
+  snapshots,
+  reload,
+  openSnapshotDetails,
+  history,
+}) => {
+  const {
+    core: {
+      i18n: { FormattedMessage, translate },
+    },
+  } = useAppDependencies();
+
+  const columns = [
+    {
+      field: 'snapshot',
+      name: translate('xpack.snapshotRestore.snapshotList.table.snapshotColumnTitle', {
+        defaultMessage: 'Snapshot',
+      }),
+      truncateText: true,
+      sortable: true,
+      render: (snapshotId: string, snapshot: SnapshotDetails) => (
+        <EuiLink onClick={() => openSnapshotDetails(snapshot.repository, snapshotId)}>
+          {snapshotId}
+        </EuiLink>
+      ),
+    },
+    {
+      field: 'repository',
+      name: translate('xpack.snapshotRestore.snapshotList.table.repositoryColumnTitle', {
+        defaultMessage: 'Repository',
+      }),
+      truncateText: true,
+      sortable: true,
+      // We deliberately don't link to the repository from here because the API request for populating
+      // this table takes so long, and navigating away by accident is a really poor UX.
```

Comment:
the request takes very little time with no erroneous repositories, so maybe we can reconsider this 😄